src/main/scala/com/kinja/amqp/AmqpClient.scala

```diff
@@ -48,17 +48,29 @@ class AmqpClient(
 		repeater.foreach(_.startSchedule(ec))
 	}
 
-	private def createProducers(): Map[String, AmqpProducer] = {
+	private def createProducers(): Map[String, AmqpProducerInterface] = {
 		configuration.exchanges.map {
-			case (name: String, params: ExchangeParameters) =>
-				name -> new AmqpProducer(
-					connection,
-					actorSystem,
-					messageStore,
-					configuration.connectionTimeOut,
-					configuration.askTimeOut,
-					logger
-				)(params, ec)
+			case (name: String, producerConfig: ProducerConfig) =>
+				val channelProvider = new ProducerChannelProvider(
+					connection, actorSystem, configuration.connectionTimeOut, producerConfig.exchangeParams
+				)
+				val producer = producerConfig.deliveryGuarantee match {
+					case DeliveryGuarantee.AtLeastOnce =>
+						new AtLeastOnceAmqpProducer(
+							producerConfig.exchangeParams.name,
+							channelProvider,
+							actorSystem,
+							messageStore,
+							configuration.askTimeOut,
+							logger
+						)(ec)
+					case _ =>
```

Comment:
You should match `DeliveryGuarantee.AtLeastOnce` here instead of `_`. Then if the enum is ever expanded to more cases you'll get a non-exhaustive pattern match warning instead of silently handling a new case.